src/components/DeveloperStudio/ToolDetails.tsx

```diff
@@ -1,12 +1,21 @@
 import { Link, useNavigate } from "react-router-dom";
 import { useParams } from "react-router-dom";
-import classNames from "classnames";
+import { useShowTool } from "hooks/reactQuery/useDeveloperTools";
 import NoToolImg from "images/placeholder-tools.png";
 import NoToolIconImg from "images/placeholder-tool-icon.png";
 import { Button } from "react-bootstrap";
 
+import PresentationIcon from "../../assets/presentation_icon.svg?react";
+import GithubIcon from "../../assets/github_icon.svg?react";
+
 function ToolDetails() {
   const { id } = useParams();
+  const {
+    data: tool,
+    isFetching: isLoadingTool,
+    error,
+    isError,
+  } = useShowTool(id);
 
   const navigate = useNavigate();
 
```

Comment:
add 
`if(isLoadingTool){
return <PageLoader />
}`
this helps user to understand the data is loading and bellow you wont have to do `tool?.` everywhere

Reply:
Checking